### PR TITLE
Project Submission

### PR DESCRIPTION
**Pull request by bruk19**
**In This PR:-**
<hr>

   >In this project I implement the required features to set up and deploy a Budget-app that can help users manage their 
     expenses professionally.

  You can learn more by clicking on the link below

  [Live Demo Link](https://www.loom.com/share/6593215a30d0483f8037916a41f57e9f)

<hr>

- Splash screen

    >A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

-  Sign up and log in pages

     >The user should be able to register in the app with full name, email and password (all mandatory).
     >The user can log into the app using email and password.
     >If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)

     >When the user logs in, they are presented with the categories page.
     >For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that 
       category.
     >When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
     >There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page

     >For a given category, the list of transactions is presented, ordered by the most recent.
     >At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions 
       in that category).
     >There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
     >When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page

     >The user fills out a form to create a new category, indicating their name and icon (both mandatory).
     >The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
     >When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page

     >The user fills out a form to create a new transaction with:
          -name (mandatory)
          -amount (mandatory)
          -categories (mandatory at least one)
    >The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that 
       category.
    >When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,80 @@
...
+- Major: Ruby, HTML,CSS
+- Frameworks: Rails, Bootstrap5
+- Libraries: Cancancan, Devise
+- Technologies: Git, Gitflow, Linters, Github, PostgreSql
 
-* Database creation
+## Live Demo (not available yet)
 
-* Database initialization
+[Live Demo Link](https://budgetify-ap.herokuapp.com/)
```
Comment (Davidosky007) [inline README.md]: * Your live version link doesn't work as when i try to signup it throws an error , Kindly fix this error 
image

File: README.md [hunk 2]
```diff
@@ -1,24 +1,80 @@
...
+Feel free to check the [issues page](https://https://github.com/bruk19/Budget-App/issues).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Hat tip to anyone whose code was used
+- Inspiration
```
Comment (Davidosky007) [inline README.md]: * Kindly ensure your README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/). as stated in the [project requirement ](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=Original%20design%20idea,of%20your%20project.)

File: db/migrate/20222025090441_add_references.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class AddReferences < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :entities, :user, foreign_key: { to_table: :users }, index: true
+    add_reference :groups, :user, foreign_key: { to_table: :users }, index: true
```
Comment (Davidosky007) [inline db/migrate/20222025090441_add_references.rb]: * According to the project requirement we are instructed not to change column names visible in the diagram. for example the foreign key for entities should be `author_id ` and not `user_id`.   you can write something like this ` t.references :author, foreign_key: { to_table: 'users' }) 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @bruk19 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* All tests passed ✔️ 
* Descriptive commit messages 👍 


## Required Changes ♻️
* Good job with the video presentation but you missed this part of the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=you%20should%20also%20highlight%20some%20interesting%20piece%20of%20code%20or%20something%20you%20built%20that%20you%20are%20very%20proud%20of.) that says you should also highlight some interesting piece of code or something you built that you are very proud of. Kindly fix this 🙏 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/schema.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  enable_extension "plpgsql"
+
+  create_table "entities", force: :cascade do |t|
+    t.string "name"
+    t.decimal "amount"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.bigint "user_id"
+    t.index ["user_id"], name: "index_entities_on_user_id"
+  end
```
Comment (brytebee) [inline db/schema.rb]: - There are no schema changes to swap `author_id` for `user_id`  as also requested previously.

image

File: README.md [hunk 5]
```diff
@@ -1,24 +1,78 @@
...
+- Major: Ruby, HTML,CSS
+- Frameworks: Rails, Bootstrap5
+- Libraries: Cancancan, Devise
+- Technologies: Git, Gitflow, Linters, Github, PostgreSql
 
-* Database creation
+## Live Demo (not available yet)
 
-* Database initialization
+[Live Demo Link](https://budget01app.herokuapp.com/)
```
Comment (brytebee) [inline README.md:18]: - You've removed the video without a replacement as requested by the previous reviewer. Kindly make the change requested and add the video link.

File: README.md [hunk 6]
```diff
@@ -1,24 +1,78 @@
...
+## Built With
 
-* Configuration
+- Major: Ruby, HTML,CSS
+- Frameworks: Rails, Bootstrap5
+- Libraries: Cancancan, Devise
+- Technologies: Git, Gitflow, Linters, Github, PostgreSql
 
-* Database creation
+## Live Demo (not available yet)
```
Comment (brytebee) [inline README.md]: - [ ] **Optional:** The link is now available, kindly remove the (not available yet) in the **Live Demo** Section title.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @bruk19👋, 

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️
- You added the live link and it works great.✔️

##

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions
-N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- 

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**